runtime/sema/check_function.go

```diff
@@ -393,7 +393,7 @@ func (checker *Checker) visitWithPostConditions(postConditions *ast.Conditions,
 			// here the `result` value in the `post` block will have type `auth(E, X, Y) &R`
 			if entitlementSupportingType, ok := innerType.(EntitlementSupportingType); ok {
 				supportedEntitlements := entitlementSupportingType.SupportedEntitlements()
-				if supportedEntitlements == nil || supportedEntitlements.Len() > 0 {
+				if supportedEntitlements != nil && supportedEntitlements.Len() > 0 {
```

Comment:
Why did this change? Do we have a test case?

Reply:
This I think is left over from a previous PR; `SupportedEntitlements` used to return `nil` and now it cannot, so these two cases are actually equivalent. I just removed the first condition to simplify.